rpc/eth_api.go

```diff
@@ -180,6 +180,21 @@ func (e *PublicEthAPI) GetStorageAt(address common.Address, key string, blockNum
 
 // GetTransactionCount returns the number of transactions at the given address up to the given block number.
 func (e *PublicEthAPI) GetTransactionCount(address common.Address, blockNum BlockNumber) (*hexutil.Uint64, error) {
+	if blockNum == LatestBlockNumber {
+		// Get nonce (sequence) from account
+		from := sdk.AccAddress(address.Bytes())
+		authclient.Codec = codec.NewAppCodec(e.cliCtx.Codec)
+		accRet := authtypes.NewAccountRetriever(authclient.Codec, e.cliCtx)
+
+		_, nonce, err := accRet.GetAccountNumberSequence(from)
+		if err != nil {
+			return nil, err
+		}
+
+		n := hexutil.Uint64(nonce)
+		return &n, nil
```

Comment:
this section should be the only content of the `GetTransactionCount` function

Reply:
the GetTransactiomCount function is supposed to also be query for past nonces at specific blocks as well, and this part only checks for the current nonce. 

do you know if there's another way to get the nonce at a certain height that would be better?

Reply:
You can set the height to the context using WithHeight

Reply:
updated